### ISSUE TITLE
[BUG] Text is cut off on bars where textposition=outside

### ISSUE THREAD

**Issue by RedShift1** [plotly/plotly.js#2001]
When drawing a bar plot with
`{ textposition: outside }`

the text of the highest values is sometimes cut off:

![plotly_bars_text](https://user-images.githubusercontent.com/3098010/30381763-a0b1197a-989d-11e7-83f4-d81cc58a8b2d.png)

JSFiddle: https://jsfiddle.net/vuqugx8v/

**Comment by alexcjohnson** [comment]
Yes, this is the same issue as #2000 - closing this one, we will make sure the fix handles both scatter and bar traces.

**Comment by babykick** [comment]
Has the bug been Fixed or not yet? I recently use Dash to plot bar chart, the cut off problem still exists.

**Comment by dfaivre** [comment]
In case some one stumbles across this, use the `cliponaxis: false` option (from the above [referenced PR](https://github.com/plotly/plotly.js/pull/2378))

**Comment by pietersv** [comment]
I just ran into a case where setting `cliponaxis: false` just wouldn't work.  One workaround is to explicitly set the yaxis range to `[0,7000]`  but I think more direct is to set the top margin to 20px or something large enough to show the label

**Set top margin to 20px**
![image](https://user-images.githubusercontent.com/10464727/141403212-34aa1de9-a8d0-4425-9c8e-05383ada7233.png)

**Set yaxis range a little wider:**
![image](https://user-images.githubusercontent.com/10464727/141403108-bad7c3d6-1b7a-476b-ac71-6e93ab732fd1.png)

**Comment by brandonerose** [comment]
for a bar chart the only thing that works for me is calculating the max value and multiplying the max by 1.1 to extend the range. Hope this gets fixed.

**Comment by brandonerose** [comment]
I found a useful workaround! If you add the label as an **annotation** instead of texposition outside it will not get cut off :)

**Comment by emilykl** [comment]
special case of https://github.com/plotly/plotly.js/issues/2000

**Comment by emilykl** [comment]
Reopening this one as it best captures the specific issue addressed by #7675
